src/chat/resourceConnection.ts

```diff
@@ -0,0 +1,152 @@
+export function isInsightsWorkspaceConnection(connection: AzureResourceConnection | undefined): connection is InsightsWorkspaceConnection {
+    return connection?.type === insightsWorkspaceType;
+}
+
+export function getConnectionTypeDisplayName(connection: AzureResourceConnection): string | undefined {
+    switch (connection.type) {
+        case kubernetesManagedClusterType:
+            return "Kubernetes Cluster";
+        case containerAppType:
+            return "Container App";
+        case appServiceType:
+            if (isWebAppConnection(connection)) {
+                return "Web App";
+            } else if (connection.kind?.startsWith("functionapp")) {
+                return "Function App";
+            } else if (connection.kind === "workflowapp") {
+                return "Logic App";
+            }
+            // Likely a "containerized" or "kubernetes" web app
+            return "App Service";
+        case insightsComponentType:
+            return "Application Insights";
+        case insightsWorkspaceType:
+            return "Log Analytics Workspace";
+        case "microsoft.resources/resourcegroups":
+            return "Resource Group";
+        case "microsoft.app/containerapps":
+            return "Container App";
+        case "microsoft.app/managedenvironments":
+            return "Container App Environment";
+        case "microsoft.compute/virtualmachines":
+            return "Virtual Machine";
+        case "microsoft.documentdb/databaseaccounts":
+            return "Cosmos DB Account";
+        case "microsoft.storage/storageaccounts":
+            return "Storage Account";
+        case "microsoft.web/staticsites":
+            return "Static Web App";
+        default:
+            return undefined;
+    }
+}
+
+export function getResourceIdFromConnection(connection: AzureResourceConnection): string {
+    return `subscriptions/${connection.subscriptionId}/resourceGroups/${connection.resourceGroup}/providers/${connection.type}/${connection.name}`;
+}
+
+export function resourceConnectionFromResourceId(resourceId: string, kind?: string, expectedType?: string): AzureResourceConnection | undefined {
+    const parts = resourceId.split("/");
+    const subscriptionId = parts[2];
+    const resourceGroup = parts[4];
+    const type = parts.slice(6, parts.length - 1).join("/").toLowerCase();
+    const name = parts[parts.length - 1];
+    if (expectedType && type !== expectedType) {
+        return undefined;
+    } else {
+        return {
+            type: type,
+            kind: kind,
+            name: name,
+            subscriptionId: subscriptionId,
+            resourceGroup: resourceGroup,
+        };
+    }
+}
+
+export async function getTokenCredentialForAzureResourceConnection(subscriptionProvider: AzureSubscriptionProvider, connection: AzureResourceConnection): Promise<{ credential: TokenCredential, tenantId: string } | undefined> {
```

Comment:
This doesn't need to be in this extension. I'll delete next week while Jason is gone.